File: C#/BluffinMuffin.Protocol/Game/PlayerTurnBeganCommand.cs
using BluffinMuffin.Protocol.DataTypes.Attributes;

namespace BluffinMuffin.Protocol.Game
{
    /// <summary>
    /// The server sends this command to indicated that it's time for a specific player to play.
    /// </summary>
    public class PlayerTurnBeganCommand : AbstractGameCommand
    {
        /// <summary>
        /// The position on the table where is sitting the player that needs to play
        /// </summary>
        [ExampleValue(7)]
        public int NoSeat { get; set; }

        /// <summary>
        /// The amount needed to call. It's the minimum to play if you don't want to fold.
        /// </summary>
        [ExampleValue(21)]
        public int AmountNeeded { get; set; }

        /// <summary>
        /// Usually True. Indicates if the player have the option to Fold. It's false when it's the bring-in.
        /// </summary>
        [ExampleValue(true)]
        public bool CanFold { get; set; }

        /// <summary>
        /// The minimum amount needed to raise. A raise must be at least the size of the largest previous bet or raise of the current betting round. To raise, the player have to send at least AmountNeeded+MinimumRaiseAmount.
        /// </summary>
        [ExampleValue(42)]
        public int MinimumRaiseAmount { get; set; }

        /// <summary>
        /// The maximum amount somebody can raise. Total, the player can put AmountNeeded+MaximumRaiseAmount. When No Maximum, this is set to int.MaxValue
        /// </summary>
        [ExampleValue(4200)]
        public int MaximumRaiseAmount { get; set; }
    }
}

File: Python/bluffinmuffin/protocol/game/player_turn_began_command.py
from bluffinmuffin.protocol.interfaces import AbstractGameCommand


class PlayerTurnBeganCommand(AbstractGameCommand):
    def __init__(self, table_id, no_seat, amount_needed, can_fold, minimum_raise_amount, maximum_raise_amount):
        super().__init__(table_id)
        self.no_seat = no_seat
        self.amount_needed = amount_needed
        self.can_fold = can_fold
        self.minimum_raise_amount = minimum_raise_amount
        self.maximum_raise_amount = maximum_raise_amount

    def __str__(self):
        return '{0} ({1} {2} {3}/{4}/{5})'.format(
            super().__str__(),
            self.no_seat,
            self.amount_needed,
            self.can_fold,
            self.minimum_raise_amount,
            self.maximum_raise_amount
        )

    def _encode_specific(self, d):
        super()._encode_specific(d)
        d['NoSeat'] = self.no_seat
        d['AmountNeeded'] = self.amount_needed
        d['CanFold'] = self.can_fold
        d['MinimumRaiseAmount'] = self.minimum_raise_amount
        d['MaximumRaiseAmount'] = self.maximum_raise_amount

    @classmethod
    def decode(cls, obj):
        return cls(
            obj["TableId"],
            obj["NoSeat"],
            obj["AmountNeeded"],
            obj["CanFold"],
            obj['MinimumRaiseAmount'],
            obj['MaximumRaiseAmount']
        )
